usage/jsgui/src/main/webapp/assets/js/view/application-add-wizard.js

```diff
@@ -85,6 +85,13 @@ define([
         if (entity.config && _.size(entity.config)) result["brooklyn.config"] = entity.config;
         return result;
     }
+    function getConvertedConfigValue(value) {
+        try {
+            return $.parseJSON(value);
+        } catch (e) {
+            return value;
```

Comment:
Theoretically a user could've declared his config key as String only to be able to use UI provided values, while treating it as an object under the hood. I wasn't able to find such cases though.

To be more robust you could check the key type and convert only if != enum, string.